dtcli/scripts/dt.py

```diff
@@ -313,18 +291,110 @@ def genca(**kwargs):
+    "-o",
+    "--output",
+    "destination",
+    type=click.Path(writable=True),
+    callback=mk_click_callback(Path),
+    required=True,
+    help="Location where the keycert will be written",
+)
+@click.option(
+    "--ca-crt",
+    type=click.Path(exists=True, readable=True, dir_okay=False),
+    callback=mk_click_callback(Path),
+    required=True,
+    help="Location of CA public certificate"
+)
+@click.option(
+    "--ca-key",
+    type=click.Path(exists=True, readable=True, dir_okay=False),
+    callback=mk_click_callback(Path),
+    required=True,
+    help="Location of CA private key"
+)
+@click.option(
+    "--name",
+    prompt=True,
+    # TODO: more restrictive validation
+    help="Name of the certificate holder, likely developer name",
+)
+@click.option(
+    "--company",
+    # TODO: more restrictive validation
+    help="Name of the company that the holder belongs to",
+)
+@click.option(
+    "--days-valid",
+    default=defaults.DEFAULT_CERT_VALIDITY,
+    show_default=True,
+    # TODO: more restrictive validation
+    type=int,
+    help="Number of days certificate will be valid",
+)
+@click.option(
+    "-f",
+    "--force",
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="Ignore subtleties, overwrite without prompt, when in doubt - advance!",
+)
+# TODO: additional subject parameters
+def generate_developer_pem(destination, ca_crt, ca_key, name, company, days_valid, force):
+    """
+    Generate a keycert for developer.
+
+    This should be signed by CA and belong to one entity only (like an employee). The resulting file is a fused key-certificate that allows to sign extensions on behalf the Certificate Authority.
+
+    Certificates with passphrase are currently not supported as if you required that kind of level of security it wouldn't be wise to use this command in it's current form.
+    """
+    if destination.exists() and not force:
+        raise click.BadParameter(f"destination {destination} already exists, please try again with --force to proceed irregardless", param_hint="--source")
+
+    subject_kv = [
+        ("CN", name),
+    ]
+    
+    if company:
+        subject_kv.append(("O", company))
+    # TODO: get additional keys - via an additional n-argument
+
+    # TODO: is this the correct format?
+    subject = "".join(map(lambda t: f"/{t[0]}={t[1]}", subject_kv))
+    # TODO: maybe I can just unparse? What about order?
+    subject=validate_parse_subject(None, None, subject)
+    # TODO: test logic after clayring that
+
+    # TODO: see sign
+    # TODO: implement sensible passphrase handling - it should be a prompt only when it's required and handled securely (like... cleared from memory), also: get rid of the comment in help
+    # TODO: both setting the dev passphrase and reading the CA key passphrase
+
+    signing.generate_cert(
+        ca_cert_file_path=ca_crt,
+        ca_key_file_path=ca_key,
+        destination=destination,
+        subject=subject,

```

Comment:
acc. line 299 dev-subject

Reply:
this is actually fine, as I build the subject above ;)